sox.features.js

```diff
@@ -35,45 +35,57 @@
+      // Description: Adds the Stack Exchange logo next to users that *ARE* Stack Exchange employees
 
-      const anchors = [...document.querySelectorAll('.comment a, .deleted-answer-info a, .employee-name a, .user-details a, .question-summary .started a')].filter(el => {
-        return !el.parentElement.classList.contains('user-gravatar32') && el.href && el.href.contains('/users/');
-      });
-      const ids = [];
+      const $icon = $('<svg aria-hidden="true" class="sox-markEmployees-logo sox-sprite svg-icon iconStackExchange" width="18" height="18" viewBox="0 0 18 18"><path d="M15 1H3a2 2 0 0 0-2 2v2h16V3a2 2 0 0 0-2-2zM1 13c0 1.1.9 2 2 2h8v3l3-3h1a2 2 0 0 0 2-2v-2H1v2zm16-7H1v4h16V6z"></path></svg>');
+      $icon.css('color', $('.mod-flair').css('color'));
 
-      for (let i = 0; i < anchors.length; i++) {
-        const href = anchors[i].href;
-        const idMatch = href.match(/(\d+)/);
-        if (idMatch && ids.indexOf(idMatch[1]) === -1) ids.push(idMatch[1]);
-      }
-      sox.debug('markEmployees user IDs', ids);
+      function getIds() {
+        const anchors = [...document.querySelectorAll('.comment a, .deleted-answer-info a, .employee-name a, .user-details a, .question-summary .started a')].filter(el => {
+          return !el.parentElement.classList.contains('user-gravatar32') && $(el).find('.sox-markEmployees-logo') && el.href && el.href.contains('/users/');
+        });
+        const ids = [];
 
-      // TODO is pagination needed?
-      sox.helpers.getFromAPI({
-        endpoint: 'users',
-        ids,
-        sitename: sox.site.url,
-        filter: '!*MxJcsv91Tcz6yRH',
-        limit: 100,
-        featureId: 'markEmployees',
-        cacheDuration: 60 * 24, // Cache for 24 hours (in minutes)
-      }, items => {
-        sox.debug('markEmployees returned data', items);
-        for (let i = 0; i < items.length; i++) {
-          const userId = items[i].user_id;
-          const isEmployee = items[i].is_employee;
-          if (!isEmployee) continue;
-
-          const $icon = $('<svg aria-hidden="true" class="sox-markEmployees-logo sox-sprite svg-icon iconStackExchange" width="18" height="18" viewBox="0 0 18 18"><path d="M15 1H3a2 2 0 0 0-2 2v2h16V3a2 2 0 0 0-2-2zM1 13c0 1.1.9 2 2 2h8v3l3-3h1a2 2 0 0 0 2-2v-2H1v2zm16-7H1v4h16V6z"></path></svg>');
-          $icon.css('color', $('.mod-flair').css('color'));
-          anchors.filter(el => el.href.contains(`/users/${userId}/`)).forEach(el => {
-            $(el).append($('<span/>', {
-              title: 'employee (added by SOX)',
-            }).append($icon.clone()));
-          });
+        for (let i = 0; i < anchors.length; i++) {
+          const idMatch = anchors[i].href.match(/(\d+)/);
+          if (idMatch && ids.indexOf(idMatch[1]) === -1) ids.push(idMatch[1]);
         }
-      });
+        sox.debug('markEmployees user IDs', ids);
+
+        for (let i = 1; i < Math.trunc(ids.length / 100) + 2; i++) {
```

Comment:
If `ids.length` is `100` here, will the loop `i < Math.trunc(ids.length / 100) + 2` condition be 3 instead of 2 which is what we'd actually want?

Reply:
A simple `for (let i = 1; i < Math.trunc(100 / 100) + 2; i++) { console.log(i); }` says `1` and `2` - note that `i` isn't `0`, but `1`.

Reply:
If we've got a page length of 100 though, do we not just want page `1` and not `2`?

Reply:
Most probably /me who thought 100 ids need 2 API calls. Will think of it tomorrow - if you have a suggestion, can you please add a comment?

Reply:
I think:
```
for (let i = 0; i < Math.ceil(ids.length / 100); i++) {
    apiCall(i + 1, ids.slice(i * 100, (i * 100) + 100), anchors);
}
```
should work fine

Reply:
My idea was:

    for (let i = 1; i < Math.round(ids.length * 100) / 10000 + 1; i++) {
      apiCall(i, ids.slice(i * 100 - 100, i * 100), anchors);
     }

Reply:
I think the `ceil` version would probably be more readable and easier to understand because it doesn't have the division by 10000 and reads a bit more naturally.

You could still start the loop at index 1 thought if you would prefer that!